src/RenderJSONVisitor.cc

```diff
@@ -0,0 +1,136 @@
+//
+//  RenderJSONVisitor.cc
+//  libdrafter
+//
+//  Created by Pavan Kumar Sunkara on 17/06/15.
+//  Copyright (c) 2015 Apiary Inc. All rights reserved.
+//
+
+#include "RenderJSONVisitor.h"
+#include "sosJSON.h"
+#include "stream.h"
+
+using namespace refract;
+
+namespace drafter
+{
+
+    RenderJSONVisitor::RenderJSONVisitor(const sos::Base::Type& type_)
+    : type(type_) {}
+
+    RenderJSONVisitor::RenderJSONVisitor()
+    : type(sos::Base::UndefinedType) {}
+
+    void RenderJSONVisitor::assign(sos::Base value) {
+        if (type == sos::Base::ArrayType) {
+            pArr.push(value);
+        }
+        else if (type == sos::Base::UndefinedType) {
+            result = value;
+        }
+    }
+
+    void RenderJSONVisitor::assign(std::string key, sos::Base value) {
+        if (!key.empty() && type == sos::Base::ObjectType) {
+            pObj.set(key, value);
+        }
+    }
+
+    void RenderJSONVisitor::visit(const IElement& e) {
+        if (e.element() == "object") {
```

Comment:
I understand. I didn't register the visitor with the Element.h because I didn't want it to be tightly coupled with `IElement`. Can we register the visitor without doing that?